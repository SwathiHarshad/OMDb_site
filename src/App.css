.App {
  background-color: #4cb7a5 ;
  min-width: 98vw;
  min-height: 100vh;
  position: relative;

}
input {
  border: none;
  outline: none;
  font-size: 20px;
  background-color: #4cb7a5;
}
.mainContent{
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
.display-flex{
  display: flex;
  flex-direction: column;
  align-items: space-evenly;
}
.mainContent > div {
  margin: 20px 15px;
}
.popup{
  position: fixed;
  top: 0px;
  width: 0vw;
  height: 100vh;
  background:rgba(0,0,0, 0.7);
  color: wheat;
  transition: width .5s;
}
.show {
  width: 100vw;
}


