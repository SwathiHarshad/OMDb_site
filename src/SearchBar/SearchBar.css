.searchBar {
  display: flex;
  justify-content: center;
  padding: 20px 0px;
}
.inputBox {
  padding: 10px;
  border-bottom: 1px solid ;
}
/* button CSS*/
.btn {
  padding: 10px 25px;
  background-color: lightgray;
  -webkit-transition: all 0.8s ease;
  -moz-transition: all 0.8s ease;
  transition: all 0.8s ease;
  margin-left: 10px;

}
.btn:hover {
  background-color: #6495ed;
}