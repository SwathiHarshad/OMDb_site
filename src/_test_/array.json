[{"Title":"James Bond 007: Everything or Nothing","Year":"2003","imdbID":"tt0366629","Type":"game","Poster":"https://m.media-amazon.com/images/M/MV5BMTk2MDYwMDU1Nl5BMl5BanBnXkFtZTgwNTM2NTQ3MDE@._V1_SX300.jpg"},{"Title":"James Bond 007: Blood Stone","Year":"2010","imdbID":"tt1692489","Type":"game","Poster":"N/A"},{"Title":"Jatt James Bond","Year":"2014","imdbID":"tt3732110","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BOWVmNjQ2YmYtY2M4ZS00NDdkLTljZmUtYTZkNmNjM2FhMWU3XkEyXkFqcGdeQXVyNjQ2MjQ5NzM@._V1_SX300.jpg"},{"Title":"James Bond 007: Goldfinger","Year":"1986","imdbID":"tt0924449","Type":"game","Poster":"https://m.media-amazon.com/images/M/MV5BZTM4N2I0NWYtNWUyYS00NjkzLWIwNWMtOGU1MWZmYWM5MzRkXkEyXkFqcGdeQXVyNzg5OTk2OA@@._V1_SX300.jpg"},{"Title":"James Bond Jr.","Year":"1991â€“1992","imdbID":"tt0283744","Type":"series","Poster":"https://m.media-amazon.com/images/M/MV5BMTQ2NjkzNTA1N15BMl5BanBnXkFtZTcwMzM1Mjg4NA@@._V1_SX300.jpg"},{"Title":"The James Bond Story","Year":"1999","imdbID":"tt0239074","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMTg1ODc0MjkzMF5BMl5BanBnXkFtZTcwMDY2NzEyMQ@@._V1_SX300.jpg"},{"Title":"James Bond in Agent Under Fire","Year":"2001","imdbID":"tt0295846","Type":"game","Poster":"N/A"},{"Title":"James Bond Supports International Women's Day","Year":"2011","imdbID":"tt1858469","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BNmQ4OTlmNTUtMmQ1NC00YmI0LTgxYTItYjU5NjJkZjFkZTMyXkEyXkFqcGdeQXVyMjExMDE1MzQ@._V1_SX300.jpg"},{"Title":"James Bond","Year":"2015","imdbID":"tt4896340","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BNWU3NzNhY2UtZDM0ZS00MmYwLWIxMDgtZGUwMzEwZDUyMGY3XkEyXkFqcGdeQXVyNjAzNzExNTk@._V1_SX300.jpg"},{"Title":"Happy Anniversary 007: 25 Years of James Bond","Year":"1987","imdbID":"tt0223426","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BNWE3NzFlZDItZTllNy00ZGE2LTgwMWYtZWI3ZGQyYjY4ZTNiXkEyXkFqcGdeQXVyMTY1MTcxMzc@._V1_SX300.jpg"}]