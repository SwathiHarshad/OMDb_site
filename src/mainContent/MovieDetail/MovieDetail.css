/* Popup display with transition */
#movieDetail.mainContent{
  align-items: center;
  height: 100vh;
  display: none;
  transition: width 2s;
  margin-left: 8%;
}
.displayBlock{
  width: 100vw;
  display: flex!important;
}
/* Close Icon */
.closeBtn{
  position: absolute;
  top: 0px;
  font-size: 45px;
  right: 30px;
  transform: rotate(45deg);
  cursor: pointer;
}
/* Content CSS */
.Details{
  width: 72%;
}
.Details > div {
  display: flex;
  margin-bottom: 10px;
}
.label {
  width: 12%;
}
.name {
  width: 86%;
}
#movieDetail .display-flex{
  color: black;
}
