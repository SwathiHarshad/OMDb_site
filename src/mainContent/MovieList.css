.imageCSS{
  width: 200px;
  height: 300px;
  background-color: #DAD9D9;
  border-radius: 12px;
  -webkit-transition: all 0.8s ease;
  -moz-transition: all 0.8s ease;
  transition: all 0.8s ease;
}
.imgDes {
  display: none;
  width: 0px;
  color: black;
  border-radius: 12px;

  justify-content: center;
  position: absolute;
  height: 350px;
  padding: 0px 20px;
  background: rgba(11, 37, 229, 0.3);
  font-size: 17pt;

  -webkit-transition: all 0.8s ease;
  -moz-transition: all 0.8s ease;
  transition: all 0.8s ease;
}
.imgDes > div {
  border-radius: 3px;
  margin-bottom: 10px; 
  color: #ffffff; 
}
.navigationIcon {
  background-image:url('../asset/arrow_circle_down-24px.svg');
  transform: rotate(-90deg);
}

/*hover css to enlighten the focused content */
.poster:hover .imageCSS {
  width: 250px;
  height: 350px;
  opacity: 0.4;
}
.poster:hover .imgDes {
  display: flex;
  width: 208px;
}
