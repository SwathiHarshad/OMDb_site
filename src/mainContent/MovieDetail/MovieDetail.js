import React, { useState, useEffect } from 'react'
import './MovieDetail.css'

export function MovieDetail({Detail, toClosePopup, toShow}){
  const [displayBlock, setDisplay] = useState('')
  useEffect(()=>{
    if(toShow === 'show'){
      setDisplay('displayBlock')
    } else{
      setDisplay()
    }
  },[toShow])
  return(
    <div id='movieDetail' className={'mainContent '+displayBlock}>
      <span className='closeBtn' onClick={toClosePopup}>+</span>
      <div className='Poster'>
        <img src={Detail.Poster} className='imageCSS' alt={Detail.Title}></img>
        
      </div>
      <div className='Details'>
        <div>
          <div className='label'>Title: </div>
          <div className='name'>{Detail.Title}</div>
        </div>
        <div>
          <div className='label'>description: </div>
          <div className='name'>{Detail.Plot}</div>
        </div>

        <div>
          <div className='label'>Runtime: </div>
          <div className='name'>{Detail.Runtime}</div>
        </div>
        <div>
          <div className='label'>Genre: </div>
          <div className='name'>{Detail.Genre}</div>
        </div>
        <div>
          <div className='label'>Actors: </div>
          <div className='name'>{Detail.Actors}</div>
        </div>
    {
      (Detail.Director !== 'N/A' && Detail.Director !== null)?
        <div>
          <div className='label'>Director: </div>
          <div className='name'>{Detail.Director}</div>
        </div>
      : null
    }
    {
      (Detail.Writer !== 'N/A' && Detail.Writer !== null)?
        <div>
          <div className='label'>Writer: </div>
          <div className='name'>{Detail.Writer}</div>
        </div>
      : null
    } 
    {
      (Detail.Production !== 'N/A' && Detail.Production !== null)?
        <div>
          <div className='label'>Production: </div>
          <div className='name'>{Detail.Production}</div>
        </div>
      : null
    }
    {
      (Detail.Production !== 'N/A' && Detail.Production !== null)?
        <div>
          <div className='label'>Awards: </div>
          <div className='name'>{Detail.Awards}</div>
        </div>
      : null
    }
        
        

        
        


      </div>
    </div>
  )
}