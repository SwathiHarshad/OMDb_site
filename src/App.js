import React,{useEffect, useState} from 'react'
import './App.css'
import SearchBar from './searchBarAndSorting/SearchBar'
import { MovieList } from './mainContent/MovieList'
import { MovieDetail } from './mainContent/MovieDetail/MovieDetail'
import { ErrorHandling } from './mainContent/MovieDetail/ErrorHandling/ErrorHandling'
import { useGET } from './fetch/useGET'

function App () {
  const [url, setURL] = useState('')
  const [isLoading, SetLoading] = useState(false)
  const [isPagination, setPagination] = useState(false)
  const [isDetail, setPopup] = useState(false)
  const [searchData, setSearchData] = useState('')
  const [pageCount, setCount] = useState(1)
  const [movieId, setMovieId] = useState('')
  const [movieListData, setMovieList] = useState('')

  /****** Custom hook to hit the API *******/
  const [movieList, movieDetail, error] = useGET(url, isPagination, isDetail, pageCount, movieId)

  useEffect(()=>{
    setMovieList(movieList)
    console.log('App.js: ', movieList)
  },[movieList])

  /** For pagination */
  useEffect(()=>{
    window.addEventListener('scroll', handleScroll);
      return () => window.removeEventListener('scroll', handleScroll);
  },[])

  /* To Increment the page count value by one when it reaches the bottom */
  function handleScroll(){
    if (window.innerHeight + document.documentElement.scrollTop >= document.documentElement.offsetHeight-10 && !isLoading ){
      setPagination(true)
      setCount(pageCount=> parseInt(pageCount)+1)
    }
  }

/******  Update the URL to trigger the API call 
 * When 
 * The search input change or 
 * Particular movie were clicked or 
 * When the user reaches the bottom  *******/
  useEffect(() => {
    if(searchData !== ''){
      SetLoading(isLoading=> true)
      let baseURL = 'http://www.omdbapi.com/?apikey=a184ae48'
      if(isDetail === false){ // For pagination and search
        setURL(baseURL + '&s=' + searchData + '&page=' + pageCount)
      } else if(isDetail === true){ // To get details for a particular movie
        setURL(baseURL + '&i=' + movieId)
      }
    }
  },[searchData, pageCount, movieId, isDetail])

/** To update the search input and invoke the popup for movie detail by updating the movie id value*/ 
  function APICall(e) {
    if(e !== undefined && e.currentTarget !== undefined){
      let value = e.currentTarget.getAttribute('value')
      setPagination(false)
      setSearchData(value)
      document.body.style.overflow = 'unset';
      setCount(pageCount=> 1)
    }else if(typeof(e) === 'string'){
      setMovieId(e)
      setPopup(true)
      document.body.style.overflow = 'hidden';
    }
  }
  
/* To close the movie detail popup */
  function toClosePopup() {
    setPopup(false)
    document.body.style.overflow = 'unset';
  }

/* To sort the movie list by ascending and descending */
  function toSort(key,order){
    if(order === 'Ascending')
      setMovieList([...movieListData].sort((a,b) => (a[key] > b[key]) ? 1 : ((b[key] > a[key]) ? -1 : 0)))
    else
      setMovieList([...movieListData].sort((a,b) => (a[key] < b[key]) ? 1 : ((b[key] < a[key]) ? -1 : 0)))
  }

  return (
    <div className='App display-flex-center'>

      <SearchBar APICall={APICall} toSort={toSort}></SearchBar>

      <div className='mainContent'>
        <MovieList Data={movieListData} APICall={APICall}></MovieList>
      </div>

      <div className={`popup ${isDetail ? 'show' : ''}`}>
        <MovieDetail Detail={movieDetail} toClosePopup={toClosePopup} toShow={isDetail}></MovieDetail>
      </div>

      <div className={` hide  ${ isLoading ? 'display-flex' : ''} `}>
        <div>Loading....</div>
        <div className='Icon loadingIcon'></div>
      </div>

      <div className={`${ (error === '')? 'hide': ''}`}>
        <ErrorHandling Error={error}></ErrorHandling>
      </div>
    </div>
  )
}

export default App
