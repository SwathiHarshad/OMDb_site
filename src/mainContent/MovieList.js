import React, { useState, useRef } from 'react'
import './MovieList.css'
import { ErrorHandling } from './MovieDetail/ErrorHandling/ErrorHandling'

export function MovieList ({Data, APICall}) {
  const valueElement = useRef()

  function toLoadDetails(e) {
      APICall(e)
  }
  
  return(
    (Data)? (
      Data.map((searchItem, index)=>(
        <div className='poster display-flex'   key={index}
         onClick={()=>toLoadDetails(searchItem.imdbID)}
         ref={valueElement}>

          {
          (searchItem.Poster !== 'N/A' && searchItem.Poster !== undefined)?
            <img src={searchItem.Poster} className='imageCSS' alt={searchItem.Title}></img>
            : <ErrorHandling Error='Poster not available'></ErrorHandling>
          }
          <div className='display-flex imgDes'>
            <div>{searchItem.Title}</div>
            <div><span>(</span>{searchItem.Year}<span>)</span></div>
            <div className='Icon navigationIcon'></div>
          </div>
        </div>
      ))
    )
    : null




    
  )
}