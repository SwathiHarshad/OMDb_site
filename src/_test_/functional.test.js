import React from "react";
import ReactDOM, { render } from "react-dom";
import { shallow, mount } from 'enzyme';

import array from './array.json'
import detail from './detail.json'
import { MovieList } from "../mainContent/MovieList";
import App from "../App";
import SearchBar from "../searchBarAndSorting/SearchBar";
import { MovieDetail } from "../mainContent/MovieDetail/MovieDetail";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

/******** Mock function for spy as sortCall and APICall ********/
const spy = jest.fn((key,order)=> 
  console.log('To sort Function Key value: ',key,' Order to sort :', order))
const APICall = jest.fn((value, component)=>{
  console.log("API call were initiated search value:", value,' Component name: ', component)
})

/******** Movie List Component Testing ********/

const movieListContainer = shallow(<MovieList Data = {array} APICall={APICall} />)
describe("<MovieList/> Component Testing",()=>{
  it("To test movie list page", ()=>{
    expect(movieListContainer.getElements()).toMatchSnapshot();
  });
  it('To check the movie onClick function', ()=>{
    movieListContainer.find('.poster').first().props().onClick()
    expect(APICall).toHaveBeenCalled()
  })
})


/******** To Test the Search bar and Sort icon *******/

const searchBarContainer = mount(  <SearchBar APICall={APICall} toSort={spy} />  )
describe("<searchBar/> to test search and sorting functionality",()=>{

  it("Sorting type select from the dropdown", ()=>{
    searchBarContainer.find('.dropDownDiv').find('#Title').simulate("click")
    expect(spy).toHaveBeenCalled()
    searchBarContainer.find('.dropDownDiv').find('#Year').simulate("click")
    expect(spy).toHaveBeenCalled()
    expect(spy.mock.calls[0][0]).toEqual('Title')
    expect(spy.mock.calls[1][0]).toEqual('Year')
  })

  it("Sort Icon to toggle from Ascending to Descending", ()=>{
    searchBarContainer.find('.sortIcon').simulate('click')
    expect(spy.mock.calls[2][1]).toEqual('Ascending')

    searchBarContainer.find('.sortIcon').simulate('click')
    expect(spy.mock.calls[3][1]).toEqual('Descending')

  })

  it("To update the search input and make the APICall function", ()=>{
    searchBarContainer.find('input[type="text"]').simulate('change',{
      target:{
        value: 'wanted',
      },
    })
    expect(searchBarContainer.find('input[type="text"]').prop('value')).toEqual('wanted');
    searchBarContainer.find('.btn').first().simulate("click")
    expect(APICall).toHaveBeenCalled()
    expect(APICall.mock.calls[1][0]).toEqual('wanted')
  })
})

/****** <MovieDetail /> component render ******/
// describe("<MovieDetail /> component render",()=>{
//   const movieDetailContainer = shallow(<MovieDetail Detail={detail}/>)
//   it("",()=>{
//     expect(movieDetailContainer.getElements()).toMatchSnapshot();
//   })
// })
