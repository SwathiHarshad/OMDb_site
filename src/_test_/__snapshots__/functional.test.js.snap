// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetail /> component render  1`] = `
Array [
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond 007: Everything or Nothing"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BMTk2MDYwMDU1Nl5BMl5BanBnXkFtZTgwNTM2NTQ3MDE@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond 007: Everything or Nothing
      </div>
      <div>
        <span>
          (
        </span>
        2003
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <ErrorHandling
      Error="Poster not available"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond 007: Blood Stone
      </div>
      <div>
        <span>
          (
        </span>
        2010
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="Jatt James Bond"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BOWVmNjQ2YmYtY2M4ZS00NDdkLTljZmUtYTZkNmNjM2FhMWU3XkEyXkFqcGdeQXVyNjQ2MjQ5NzM@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        Jatt James Bond
      </div>
      <div>
        <span>
          (
        </span>
        2014
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond 007: Goldfinger"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BZTM4N2I0NWYtNWUyYS00NjkzLWIwNWMtOGU1MWZmYWM5MzRkXkEyXkFqcGdeQXVyNzg5OTk2OA@@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond 007: Goldfinger
      </div>
      <div>
        <span>
          (
        </span>
        1986
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond Jr."
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BMTQ2NjkzNTA1N15BMl5BanBnXkFtZTcwMzM1Mjg4NA@@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond Jr.
      </div>
      <div>
        <span>
          (
        </span>
        1991–1992
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="The James Bond Story"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BMTg1ODc0MjkzMF5BMl5BanBnXkFtZTcwMDY2NzEyMQ@@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        The James Bond Story
      </div>
      <div>
        <span>
          (
        </span>
        1999
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <ErrorHandling
      Error="Poster not available"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond in Agent Under Fire
      </div>
      <div>
        <span>
          (
        </span>
        2001
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond Supports International Women's Day"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BNmQ4OTlmNTUtMmQ1NC00YmI0LTgxYTItYjU5NjJkZjFkZTMyXkEyXkFqcGdeQXVyMjExMDE1MzQ@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond Supports International Women's Day
      </div>
      <div>
        <span>
          (
        </span>
        2011
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BNWU3NzNhY2UtZDM0ZS00MmYwLWIxMDgtZGUwMzEwZDUyMGY3XkEyXkFqcGdeQXVyNjAzNzExNTk@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond
      </div>
      <div>
        <span>
          (
        </span>
        2015
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="Happy Anniversary 007: 25 Years of James Bond"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BNWE3NzFlZDItZTllNy00ZGE2LTgwMWYtZWI3ZGQyYjY4ZTNiXkEyXkFqcGdeQXVyMTY1MTcxMzc@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        Happy Anniversary 007: 25 Years of James Bond
      </div>
      <div>
        <span>
          (
        </span>
        1987
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
]
`;

exports[`<MovieList/> Component Testing To test movie list page 1`] = `
Array [
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond 007: Everything or Nothing"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BMTk2MDYwMDU1Nl5BMl5BanBnXkFtZTgwNTM2NTQ3MDE@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond 007: Everything or Nothing
      </div>
      <div>
        <span>
          (
        </span>
        2003
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <ErrorHandling
      Error="Poster not available"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond 007: Blood Stone
      </div>
      <div>
        <span>
          (
        </span>
        2010
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="Jatt James Bond"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BOWVmNjQ2YmYtY2M4ZS00NDdkLTljZmUtYTZkNmNjM2FhMWU3XkEyXkFqcGdeQXVyNjQ2MjQ5NzM@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        Jatt James Bond
      </div>
      <div>
        <span>
          (
        </span>
        2014
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond 007: Goldfinger"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BZTM4N2I0NWYtNWUyYS00NjkzLWIwNWMtOGU1MWZmYWM5MzRkXkEyXkFqcGdeQXVyNzg5OTk2OA@@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond 007: Goldfinger
      </div>
      <div>
        <span>
          (
        </span>
        1986
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond Jr."
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BMTQ2NjkzNTA1N15BMl5BanBnXkFtZTcwMzM1Mjg4NA@@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond Jr.
      </div>
      <div>
        <span>
          (
        </span>
        1991–1992
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="The James Bond Story"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BMTg1ODc0MjkzMF5BMl5BanBnXkFtZTcwMDY2NzEyMQ@@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        The James Bond Story
      </div>
      <div>
        <span>
          (
        </span>
        1999
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <ErrorHandling
      Error="Poster not available"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond in Agent Under Fire
      </div>
      <div>
        <span>
          (
        </span>
        2001
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond Supports International Women's Day"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BNmQ4OTlmNTUtMmQ1NC00YmI0LTgxYTItYjU5NjJkZjFkZTMyXkEyXkFqcGdeQXVyMjExMDE1MzQ@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond Supports International Women's Day
      </div>
      <div>
        <span>
          (
        </span>
        2011
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="James Bond"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BNWU3NzNhY2UtZDM0ZS00MmYwLWIxMDgtZGUwMzEwZDUyMGY3XkEyXkFqcGdeQXVyNjAzNzExNTk@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        James Bond
      </div>
      <div>
        <span>
          (
        </span>
        2015
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
  <div
    className="poster display-flex"
    onClick={[Function]}
  >
    <img
      alt="Happy Anniversary 007: 25 Years of James Bond"
      className="imageCSS"
      src="https://m.media-amazon.com/images/M/MV5BNWE3NzFlZDItZTllNy00ZGE2LTgwMWYtZWI3ZGQyYjY4ZTNiXkEyXkFqcGdeQXVyMTY1MTcxMzc@._V1_SX300.jpg"
    />
    <div
      className="display-flex imgDes"
    >
      <div>
        Happy Anniversary 007: 25 Years of James Bond
      </div>
      <div>
        <span>
          (
        </span>
        1987
        <span>
          )
        </span>
      </div>
      <div
        className="Icon navigationIcon"
      />
    </div>
  </div>,
]
`;
